File: Task 1/Task1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Function to identify whether each line is a comment or not
void line_identifier(const char* filename) {
    FILE* file = fopen(filename, "r");
    if (file == NULL) {
        perror("Error opening file");
        exit(EXIT_FAILURE);
    }

    char line[1000];
    int line_number = 0;

    while (fgets(line, sizeof(line), file) != NULL) {
        line_number++;
        char* stripped_line = strtok(line, "\r\n");

        if (stripped_line != NULL && (stripped_line[0] == '#' || (stripped_line[0] == '/' && stripped_line[1] == '/'))) {
            printf("Line %d: Comment - %s\n", line_number, stripped_line);
        } else {
            printf("Line %d: Code - %s\n", line_number, stripped_line);
        }
    }
    fclose(file);
}

int main() {
    const char* filename = "code.txt";
    line_identifier(filename);
    return 0;
}

File: Task 1/Task1.py
def line_identifier(lines):
    result = []

    for line in lines:

        stripped_line = line.strip()

        if stripped_line.startswith('#') :
            result.append("Comment")
        else:
            result.append("Not a Comment")

    return result

filename = "code.txt"
with open(filename, "r") as file:

    lines = file.readlines()
line_types = line_identifier(lines)
for i, line in enumerate(lines):

    print(f"Line {i + 1}: {line_types[i]} - {line.strip()}")